docs/compose-dynamic-conditions.md

```diff
@@ -0,0 +1,21 @@
+# Compose Dynamic Conditions
+
+Sometimes there is a need to give an end-user a way of composing conditions under which target objects should be filtered and returned to the client. For example, "the product should be available to the shopper only if he is more than 25 years old". Simplifying condition expressions building for the end-user is often a challenge.
+
+The most common way to build condition expressions is by using a visual UI. That way the user builds the condition expression as a regular sentence. While reading a regular sentence it is easy to understand the specific circumstances the result will yield: object returned. In VirtoCommerce, we decided to implement this approach. Based on our experience gained developing 1.x version of VirtoCommerce that is implemented using WPF we decided to migrate conditions composing functionality from there. .NET Expressions fits well to filter objects and save the filter as they are serializable. For showing conditions to a user we used AngularJS as the platform uses it. So, our solution is to compose conditions using lambda expression tree and angularjs, to show the conditions to the end-user.
```

Comment:
Remove "Based on our experience gained developing 1.x version of VirtoCommerce that is implemented using WPF we decided to migrate conditions composing functionality from there."